homeassistant/components/sonos/speaker.py

```diff
@@ -593,8 +591,9 @@ async def async_check_activity(self, now: datetime.datetime) -> None:
 
     async def async_offline(self) -> None:
         """Handle removal of speaker when unavailable."""
-        async with self._subscription_lock:
-            await self._async_offline()
+        if self._subscription_lock:
+            async with self._subscription_lock:
+                await self._async_offline()
```

Comment:
Since `self._subscription_lock` will always be set when the method is called, it's better to add an `assert` instead of the nested if. The `if` has the potential to fail silently, whereas the `assert` doesn't.
```py
        assert self._subscription_lock is not None
        async with self._subscription_lock:
            await self._async_offline()
```